frontend/src/pages/home/components/sign-in-form/sign-in-form.js

```diff
@@ -1,53 +1,107 @@
-import React from "react";
-import { useQuery } from "~/models/reactUtils";
+import React, { forwardRef, useContext, useState } from "react";
+import { useForm } from "react-hook-form";
+import * as yup from "yup";
+import { StoreContext } from "~/models/reactUtils";
 
-export default function SignInForm() {
-  const { data, error } = useQuery(store => store.queryWhoAmI());
-  if (error) return <div>{JSON.stringify(error, null, 4)}</div>;
+const Field = forwardRef(({
+  name,
+  type,
+  label,
+  placeholder,
+  errors
+}, ref) => {
+  return (
+    <div className="mb-4">
+      <label
+        className="block text-gray-700 text-sm font-bold mb-2"
+        htmlFor={name}>
+        {label}
+      </label>
+      <input
+        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+        id={name}
+        name={name}
+        type={type}
+        placeholder={placeholder}
+        ref={ref}
+      />
+      {errors[name] && (
+        <p className="text-red-500 text-xs italic mt-1">
+          {errors[name].message}
+        </p>
+      )}
+    </div>
+  );
+});
 
+const SignInSchema = yup.object().shape({
+  email: yup.string().email().required(),
+  password: yup.string().min(3).required(),
+});
 
+export default function SignInForm() {
+  const store = useContext(StoreContext);
+  const { register, handleSubmit, errors } = useForm({
+    validationSchema: SignInSchema
+  });
+  const [response, setResponse] = useState({});
+  const onSubmit = async (data, e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    store.mutateLogin({
```

Comment:
I didn't use `useQuery` with `setQuery` since there is still the issue with that
https://github.com/mobxjs/mst-gql/issues/173#issuecomment-586877782
(it doesn't catch server graphql exceptions)